server/handlers/policy_relationship_handler.go

```diff
@@ -0,0 +1,128 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/layer5io/meshery/server/models"
+
+	"github.com/ghodss/yaml"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/sirupsen/logrus"
+)
+
+// Define a simple policy struct
+type Policy struct {
+	Rules string `json:"rules"`
+}
+
+type RelationObject struct {
+	DestinationID   string                 `json:"destination_id,omitempty"`
+	DestinationName string                 `json:"destination_name,omitempty"`
+	SourceId        string                 `json:"source_id,omitempty"`
+	SourceName      string                 `json:"source_name,omitempty"`
+	Port            map[string]interface{} `json:"port,omitempty"`
+}
+
+type NetworkPolicyRegoResponse struct {
+	ServicePodRelationships        []RelationObject `json:"service_pod_relationships,omitempty"`
+	ServiceDeploymentRelationships []RelationObject `json:"service_deployment_relationships,omitempty"`
+}
+
+// PolicyRelationshipRegoHandler gets the designFile in the bytes of Yaml and resolves the network relationship as of now.
+func PolicyRelationshipRegoHandler(ctx context.Context, designFile []byte) (*NetworkPolicyRegoResponse, error) {
+	// Load the policy
+	policyFile, err := ioutil.ReadFile("../meshmodel/policies/network-policy.rego") // absolute path, needs to be changed
+	if err != nil {
+		logrus.Fatal("error reading rego file", err.Error())
+		return nil, err
+	}
+
+	// Initialize the policy
+	policy := Policy{string(policyFile)}
+
+	// Initialize the rego engine with the policy
+	engine, err := rego.New(
+		rego.Query("data.network_policy"),
```

Comment:
Also take these queries out in a variable above with proper comment explaining the usage of the query.

Reply:
Or even better. Make the query configurable from consumer side and here pass it as an argument. As this function will move to Meshkit. It should contain more functionality and least configuration. Configuration should be passed from Meshery server handler.

Reply:
meshkit function https://github.com/meshery/meshkit/pull/311 is most general in nature now, thanks for the suggestion. Until, there is the support for configurability in the user's hand, its better if we would leave the query on server side, because if goes to client, it will have other concerns. right, now, the server is setup in a manner that it will run all the queries against all the policies. We will be moving to some more extent of configurability this week or later.